quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/ConfigKeystorePropertyMappers.java

```diff
@@ -38,7 +37,7 @@ public static PropertyMapper<?>[] getConfigKeystorePropertyMappers() {
         };
     }
 
-    private static Optional<String> validatePath(Optional<String> option, ConfigSourceInterceptorContext context) {
+    private static String validatePath(String option, ConfigSourceInterceptorContext context) {
```

Comment:
> Even further consideration - are there any major benefits of using the ConfigSourceInterceptorContext, instead of obtaining values from the Configuration class?

My assumption has been that ConfigSourceInterceptorContext is either there for performance or giving more control over skipping additional property mapping. However yes in practice it's much more understandable to just go directly via the Configuration. The only other drawback / consideration is that the Configuration access is static - the ConfigurationTest logic does account for that, but it's cumbersome / fragile. Using the context might also have been a way to avoid the static state.

@pedroigor can you comment on why ConfigSourceInterceptorContext was made available initially, and if it still makes sense?

Reply:
> The only other drawback / consideration is that the Configuration access is static - the ConfigurationTest logic does account for that, but it's cumbersome / fragile. Using the context might also have been a way to avoid the static state.

@shawkins Could you please elaborate a little bit your concern about the static state? AFAIK, we rely on the static state of MP/SmallRye config anyway, and the `Configuration` is only a facade made above it - we do not handle any other static state there. 

Or you mean the static state about the changing context like disabled props, runtime/buildtime props? 

--------------------------

Like, for now, we can just leave it as is and possibly provide another overloaded `transformer()` and `mapFrom` build method without the context? It's supplied by SmallRye config anyway, right? I think the DevEx might be improved a lot. Or any concern with this?

```
public Builder<T> transformer(Function<String, String> mapper) {
```

Reply:
> Could you please elaborate a little bit your concern about the static state?

It makes the logic more testable in a standalone manner, rather than needing something like the setup done in the ConfigurationTest.

> Or any concern with this?

I'd just want more clarity on the thinking behind the initial usage of the ConfigSourceInterceptorContext (and its continued usage in ExportPropertyMapper, DatabasePropertyMapper, etc.) - if it doesn't make sense anymore, then I'm fine with removing it completely rather than having a mixture of context.proceed and Configuration methods.

Reply:
> @pedroigor can you comment on why ConfigSourceInterceptorContext was made available initially, and if it still makes sense?

@pedroigor Do you remember?

Reply:
The main reason is that transformers are basically extending our interceptor behavior when processing configuration values so it makes sense for me to expose it. Also, if you call `Configuration` you'll be going through the interceptors chain again rather than resolve values from the current chain.